xclim/utils.py

```diff
@@ -286,7 +286,11 @@ def convert_units_to(
 
         tu_u = pint2cfunits(tu)
         with units.context(context or "none"):
-            out = units.convert(source, fu, tu)
+            out = xr.DataArray(
+                data=units.convert(source.values, fu, tu),
+                coords=source.coords,
+                attrs=source.attrs,
+            )
```

Comment:
I don't think this is a problem in `xclim` for now, but this doesn't copy the `name` of the input DataArray. May be adding `name=source.name` would be safer?

Reply:
Good catch. Done.